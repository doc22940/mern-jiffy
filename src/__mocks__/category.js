module.exports = {
  title: 'actions',
  data: [
    {
      gif: {
        images: {
          original: {
            url: 'https://media0.giphy.com/media/YqoQ9wgKBCNVe/giphy.gif',
            webp: 'https://media0.giphy.com/media/YqoQ9wgKBCNVe/giphy.webp',
          },
        },
        slug: 'selena-gomez-crying-break-up-YqoQ9wgKBCNVe',
      },
      name: 'breaking up',
    },
  ],
};
