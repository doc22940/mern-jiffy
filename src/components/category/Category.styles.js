import styled from 'styled-components';

import { Link } from 'react-router-dom';

export const CategoryContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(260px, 1fr));
  grid-gap: 10px;
  margin-bottom: var(--big);
  @media (min-width: );
`;

export const SubCategoryContainer = styled(Link)`
  position: relative;
  color: var(--white);
  img {
    width: 100%;
    height: 200px;
    object-fit: cover;
  }
  h4 {
    margin: 0 10px;
    font-family: var(--Merriweather);
    font-size: var(--mid);
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
    text-align: center;
    z-index: 1;
  }
`;
