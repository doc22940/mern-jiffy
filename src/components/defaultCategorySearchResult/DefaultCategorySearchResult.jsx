import React, { useContext } from 'react';
import { v4 as uuidv4 } from 'uuid';
import LazyLoad from 'react-lazyload';

import { GifContext } from '../../context/GifContext';

import Placeholder from '../placeholder/Placeholder';
import {
  CategoryContainer,
  SubCategoryContainer,
} from '../category/Category.styles';
import { Title } from '../title/Title.styles';

const DefaultCategorySearchResult = ({ type, visible }) => {
  const { searchCategory, getGif } = useContext(GifContext);
  return (
    <>
      <Title>
        {type.title && type.title} <span>GIFs</span>
      </Title>
      <CategoryContainer>
        {type.data &&
          Object.values(type.data).map((topic) => (
            <SubCategoryContainer
              to={
                visible
                  ? `/search/${topic.name.split(' ').join('-')}`
                  : `/${topic.slug}`
              }
              onClick={
                visible
                  ? () => searchCategory(`${topic.name}`)
                  : () => getGif(topic)
              }
              key={uuidv4()}
            >
              {visible && <h4>{topic.name.toUpperCase()}</h4>}
              <LazyLoad
                height={200}
                offset={100}
                placeholder={<Placeholder />}
                once
              >
                <img
                  src={
                    visible
                      ? topic.gif.images.original.webp
                      : topic.images.original.webp
                  }
                  style={{ opacity: '0.8' }}
                  alt={topic.title}
                />
              </LazyLoad>
            </SubCategoryContainer>
          ))}
      </CategoryContainer>
    </>
  );
};

export default DefaultCategorySearchResult;
