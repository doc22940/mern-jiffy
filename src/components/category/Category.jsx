import React, { useContext } from 'react';

import { GifContext } from '../../context/GifContext';

import DefaultCategorySearchResult from '../defaultCategorySearchResult/DefaultCategorySearchResult';

const Category = () => {
  const { category } = useContext(GifContext);
  return (
    <>
      <DefaultCategorySearchResult type={category} visible={true} />
    </>
  );
};

export default Category;
