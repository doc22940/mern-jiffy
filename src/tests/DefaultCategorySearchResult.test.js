import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Enzyme, { mount, shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import category from '../__mocks__/category';

import GifContextProvider, { GifContext } from '../context/GifContext';
import DefaultCategorySearchResult from '../components/defaultCategorySearchResult/DefaultCategorySearchResult';
import {
  CategoryContainer,
  SubCategoryContainer,
} from '../components/category/Category.styles';

Enzyme.configure({ adapter: new EnzymeAdapter() });

let mockProps;
let wrapper;
let mockCategory;

describe('Category should render without error and get correct props', () => {
  beforeEach(() => {
    mockCategory = {
      title: 'actions',
    };
    mockProps = {
      type: mockCategory,
      visible: true,
    };
    wrapper = mount(
      <GifContextProvider>
        <DefaultCategorySearchResult {...mockProps} />
      </GifContextProvider>
    );
  });

  test('Should render without error', () => {
    expect(wrapper.length).toBe(1);
  });

  test('Should get the correct props', () => {
    expect(wrapper.children().props().type).toBe(mockCategory);
    expect(wrapper.children().props().visible).toBe(true);
  });
});

describe('Category should get and map data correctly, match the snapshot', () => {
  beforeEach(() => {
    mockProps = {
      type: category,
      visible: true,
    };
    wrapper = mount(
      <Router>
        <GifContextProvider>
          <DefaultCategorySearchResult {...mockProps} />
        </GifContextProvider>
      </Router>
    );
  });

  test('Should show category title', () => {
    expect(wrapper.find('h1').text()).toContain(category.title);
  });
  test('Should map the category data correctly', () => {
    expect(wrapper.find(CategoryContainer).length).toBe(
      Object.keys(category.data).length
    );
  });
  // test('Should get the gif to show in individual gif page', () => {

  //   const shallowWrapper = shallow(
  //     <SubCategoryContainer to={`${category.data[0].gif.slug}`} />
  //   );
  //   console.log(wrapper.instance());
  // });
  test('Should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
