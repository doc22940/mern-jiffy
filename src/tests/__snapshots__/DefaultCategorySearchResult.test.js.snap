// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should get and map data correctly Should match the snapshot 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <GifContextProvider>
      <DefaultCategorySearchResult type={{...}} visible={true}>
        <styled.h1>
          <h1 className=\\"sc-AxhCb gBAnb\\">
            actions
             
            <span>
              GIFs
            </span>
          </h1>
        </styled.h1>
        <styled.div>
          <div className=\\"sc-AxjAm iOXpI\\">
            <Styled(Link) to=\\"/search/breaking-up\\" onClick={[Function]}>
              <Link to=\\"/search/breaking-up\\" onClick={[Function]} className=\\"sc-AxirZ hipYxo\\">
                <LinkAnchor onClick={[Function]} className=\\"sc-AxirZ hipYxo\\" href=\\"/search/breaking-up\\" navigate={[Function: navigate]}>
                  <a className=\\"sc-AxirZ hipYxo\\" href=\\"/search/breaking-up\\" onClick={[Function: onClick]}>
                    <h4>
                      BREAKING UP
                    </h4>
                    <LazyLoad height={200} offset={100} placeholder={{...}} once={true} overflow={false} resize={false} scroll={true} unmountIfInvisible={false}>
                      <span className=\\"lazyload-wrapper\\">
                        <Placeholder>
                          <styled.div>
                            <div className=\\"sc-AxiKw eELaVs\\" />
                          </styled.div>
                        </Placeholder>
                      </span>
                    </LazyLoad>
                  </a>
                </LinkAnchor>
              </Link>
            </Styled(Link)>
          </div>
        </styled.div>
      </DefaultCategorySearchResult>
    </GifContextProvider>
  </Router>
</BrowserRouter>"
`;

exports[`Category should get and map data correctly, match the snapshot Should match the snapshot 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <GifContextProvider>
      <DefaultCategorySearchResult type={{...}} visible={true}>
        <styled.h1>
          <h1 className=\\"sc-AxhCb gBAnb\\">
            actions
             
            <span>
              GIFs
            </span>
          </h1>
        </styled.h1>
        <styled.div>
          <div className=\\"sc-AxjAm iOXpI\\">
            <Styled(Link) to=\\"/search/breaking-up\\" onClick={[Function]}>
              <Link to=\\"/search/breaking-up\\" onClick={[Function]} className=\\"sc-AxirZ hipYxo\\">
                <LinkAnchor onClick={[Function]} className=\\"sc-AxirZ hipYxo\\" href=\\"/search/breaking-up\\" navigate={[Function: navigate]}>
                  <a className=\\"sc-AxirZ hipYxo\\" href=\\"/search/breaking-up\\" onClick={[Function: onClick]}>
                    <h4>
                      BREAKING UP
                    </h4>
                    <LazyLoad height={200} offset={100} placeholder={{...}} once={true} overflow={false} resize={false} scroll={true} unmountIfInvisible={false}>
                      <span className=\\"lazyload-wrapper\\">
                        <Placeholder>
                          <styled.div>
                            <div className=\\"sc-AxiKw eELaVs\\" />
                          </styled.div>
                        </Placeholder>
                      </span>
                    </LazyLoad>
                  </a>
                </LinkAnchor>
              </Link>
            </Styled(Link)>
          </div>
        </styled.div>
      </DefaultCategorySearchResult>
    </GifContextProvider>
  </Router>
</BrowserRouter>"
`;
